src/components/NavBar/BasketSideMenu.tsx

```diff
@@ -0,0 +1,74 @@
+import { useNavigate } from "react-router-dom";
+import { BasketItem, useBasket } from "../../context/useBasket";
+import { formatCurrency } from "../../utils/formatCurrency";
+import BasketSideMenuItem from "./BasketSideMenuItem";
+import "../../styles/basket.css";
+import { IoIosCloseCircleOutline } from "react-icons/io";
+
+type BasketSideMenuProps = {
+  onClose: () => void;
+  isBasketOpen: boolean;
+};
+
+const BasketSideMenu = ({ onClose, isBasketOpen }: BasketSideMenuProps) => {
+  const { basket } = useBasket();
+
+  const total = basket.reduce((acc: number, item: BasketItem) => {
+    return acc + item.quantity * item.product.price;
+  }, 0);
+
+  const navigate = useNavigate();
+
+  const handleGoToCheckout = () => {
+    onClose();
+    navigate("/checkout");
+  };
+
+  return (
+    <nav className={`basket-menu ${isBasketOpen ? "visible" : ""}`}>
+      <section
+        className={`not-menu ${isBasketOpen ? "opaque" : ""}`}
+        onClick={() => onClose()}
+      ></section>
+      <section className={`menu ${isBasketOpen ? "active" : ""}`}>
+        <div className="close-icon-wrapper" onClick={onClose}>
+          <IoIosCloseCircleOutline className="close-icon" />
+        </div>
+
+        {basket.length === 0 ? (
```

Comment:
Nitpick: this gets hard to read, it would _probably_ be more readable if you use the fall-through approach here.

So check if basket is empty, if it is render that HTML if basket is not empty render other HMTL. So you would have 2 blocks for eg. --> `{ basket.length === 0 && <>...</> }`